azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java

```diff
@@ -0,0 +1,287 @@
+            "  ,           # Comma separates additional values. \n" +
+            "  [0-9]+      # Integer of extra value (required). \n" +
+            "  (?:         # Range for extra value (optional).  \n" +
+            "    -         # Dash separates range integer.      \n" +
+            "    [0-9]+    # Range integer of extra value.      \n" +
+            "  )?          # Range for extra value (optional).  \n" +
+            ")*            # Zero or more additional values.    \n" +
+            "$             # Anchor to end of string.           ", Pattern.COMMENTS);
+
+    private static final Pattern REGEX_NEXT_INTEGER_RANGE = Pattern.compile("# extract next integers/integer range value.    \n" +
+            "([0-9]+)      # $1: 1st integer (Base).         \n" +
+            "(?:           # Range for value (optional).     \n" +
+            "  -           # Dash separates range integer.   \n" +
+            "  ([0-9]+)    # $2: 2nd integer (Range)         \n" +
+            ")?            # Range for value (optional). \n" +
+            "(?:,|$)       # End on comma or string end.", Pattern.COMMENTS);
+    private static final Pattern ANY_STRING_REGEX = Pattern.compile(".*");
+
+    private BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
+
+    public String promoteString(String name, String defaultValue, String regex, boolean isRequired) throws IOException {
+        final boolean hasDefaultValue = StringUtils.isNotBlank(defaultValue);
+        String hintMessage = hasDefaultValue ? TextUtils.green(defaultValue) : "";
+        if (!isRequired && hasDefaultValue) {
+            hintMessage = hintMessage + ", press '" + EMPTY_REPLACEMENT + "' to input empty";
+        }
+        if (StringUtils.isNotBlank(hintMessage)) {
+            hintMessage = "(" + hintMessage + ")";
+        }
+        System.out.printf("Please input the %s%s: ", name, hintMessage);
+        System.out.flush();
+        final String finalHintMessage = hintMessage;
+        final Pattern pattern = StringUtils.isNotBlank(regex) ? Pattern.compile(regex) : ANY_STRING_REGEX;
+        return loopInput(defaultValue, hasDefaultValue, isRequired, String.format("Please input the %s%s: ", name, hintMessage), input -> {
+            if (!isRequired && StringUtils.equals(EMPTY_REPLACEMENT, input.trim())) {
+                return "";
+            }
+            if (pattern.matcher(input).find()) {
+                return input;
+            }
+            throw new UserInputException(String.format(PROMPT_MESSAGE_STRING_TEMPLATE, input, name, finalHintMessage));
+        });
+    }
+
+    public Boolean promoteYesNo(Boolean defaultValue, String message, boolean isRequired) throws IOException {
+        final boolean hasDefaultValue = defaultValue != null;
+
+        if (!isRequired && hasDefaultValue) {
+            throw new IllegalArgumentException("There is no way to input empty value for a non-required field with default value.");
```

Comment:
Seems we could remove the `isRequired` for we don't allow null value, and there will be no differences between required value and not required.